integration/networking/port_mapping_test.go

```diff
@@ -0,0 +1,290 @@
+			bridgeOpts: []func(*types.NetworkCreate){
+				network.WithIPv6(),
+				network.WithIPAM("fdf1:a844:380c:b237::/64", "fdf1:a844:380c:b237::1"),
+			},
+			skipMsg: "This test never passes",
+		},
+		{
+			name:       "IPv6 - with local IP address",
+			clientAddr: getIfaceAddress(t, "eth0", true),
+			bridgeOpts: []func(*types.NetworkCreate){
+				network.WithIPv6(),
+				network.WithIPAM("fdf1:a844:380c:b247::/64", "fdf1:a844:380c:b247::1"),
+			},
+			skipMsg: "This test never passes",
+		},
+	}
+
+	tester := func(t *testing.T, d *daemon.Daemon, c *client.Client, tcID int, tc natFromLocalHostTC) {
+		ctx := context.Background()
+
+		// Sending and receiving some data is an important part of the test. If we don't do that, the TCP handshake
+		// might succeed by connecting to something else than the "server" container used below. This could happen for
+		// instance when one of the bridge subnet is used by another interface on the host (and there might be other
+		// cases). Thus, this is preventing false positives.
+		msg := "hello world"
+		serverPort := 1234 + tcID
+		serverCmd := fmt.Sprintf("echo %q | nc -l -p %d", msg, serverPort)
+
+		bridgeName := fmt.Sprintf("nat-lo-%d", tcID)
+		network.CreateNoError(ctx, t, c, bridgeName, append(tc.bridgeOpts,
+			network.WithDriver("bridge"),
+			network.WithOption("com.docker.network.bridge.name", bridgeName))...)
+		defer network.RemoveNoError(ctx, t, c, bridgeName)
+
+		ctrName := sanitizeCtrName(t.Name() + "-server")
+		publishSpec := fmt.Sprintf("%d:%d", serverPort, serverPort)
+		ctr1 := container.Run(ctx, t, c,
+			container.WithName(ctrName),
+			container.WithImage("busybox:latest"),
+			container.WithPublishedPorts(container.MustParsePortSpec(t, publishSpec)),
+			container.WithCmd("/bin/sh", "-c", serverCmd),
+			container.WithNetworkMode(bridgeName))
+		defer c.ContainerRemove(ctx, ctr1, types.ContainerRemoveOptions{
+			Force: true,
+		})
+
+		poll.WaitOn(t, container.IsInState(ctx, c, ctrName, "running"), poll.WithDelay(100*time.Millisecond))
+
+		dialer := &net.Dialer{
+			Timeout: 3 * time.Second,
+		}
+		conn, err := dialer.Dial("tcp", net.JoinHostPort(tc.clientAddr.String(), strconv.Itoa(serverPort)))
+		assert.NilError(t, err)
+		defer conn.Close()
+
+		data, err := io.ReadAll(conn)
+		assert.NilError(t, err)
+		assert.Check(t, is.Equal(msg, strings.TrimSpace(string(data))))
+	}
+
+	for flagID, flag := range []string{"--userland-proxy=true", "--userland-proxy=false"} {
+		t.Run(flag, func(t *testing.T) {
+			d := daemon.New(t)
+			d.StartWithBusybox(t, "--experimental", "--ip6tables", flag)
+			defer d.Stop(t)
+
+			c := d.NewClientT(t)
+			defer c.Close()
+
+			for tcID, tc := range testcases {
+				// tcID is made unique across all t.Run() to make sure bridge names are unique.
+				tcID = flagID*len(testcases) + tcID
+
+				t.Run(tc.name, func(t *testing.T) {
+					skip.If(t, tc.skipMsg != "", tc.skipMsg)
+					tester(t, d, c, tcID, tc)
```

Comment:
Was the `tester` func just to not nest deeper here, or was there a specific reason to not inline it here? (as it appears to only be used here).